modules/broker-auth/src/main/java/io/ballerina/messaging/broker/auth/AuthManager.java

```diff
@@ -103,7 +108,7 @@ private void registerSaslServers() {
     public SaslServer createSaslServer(String hostName, String mechanism) throws SaslException {
         SaslServerBuilder saslServerBuilder = saslMechanisms.get(mechanism);
         if (saslServerBuilder != null) {
-            SaslServer saslServer = Sasl.createSaslServer(mechanism, BrokerAuthConstants.AMQP_PROTOCOL_IDENTIFIER,
+            SaslServer saslServer = Sasl.createSaslServer(mechanism, AMQP_PROTOCOL_IDENTIFIER,
                                                           hostName,
                                                           saslServerBuilder.getProperties(),
                                                           saslServerBuilder.getCallbackHandler());
```

Comment:
We do not need to set callback handler to plain sasl server since username and password retrieved from the sasl security response. There are some sasl mechanisms which may require external callback handler to retrieve data in addition sasl security data.